hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/initialization/BlocklistAccountCreator.java

```diff
@@ -0,0 +1,169 @@
+import com.hedera.node.app.service.mono.legacy.core.jproto.JKey;
+import com.hedera.node.app.service.mono.state.migration.HederaAccount;
+import com.hedera.node.app.service.mono.utils.EntityNum;
+import com.hederahashgraph.api.proto.java.AccountID;
+import com.hederahashgraph.api.proto.java.Key;
+import com.hederahashgraph.api.proto.java.KeyList;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.bouncycastle.asn1.sec.SECNamedCurves;
+import org.bouncycastle.crypto.params.ECDomainParameters;
+import org.bouncycastle.crypto.params.ECPublicKeyParameters;
+import org.bouncycastle.util.encoders.Hex;
+
+@Singleton
+public class BlocklistAccountCreator {
+    public static final String BLOCKLIST_ACCOUNT_MEMO = "Account is blocked";
+    private static final int ZERO_BALANCE = 0;
+    private final Supplier<HederaAccount> accountSupplier;
+    private final EntityIdSource ids;
+    private final BackingStore<AccountID, HederaAccount> accounts;
+    private final Supplier<JEd25519Key> genesisKeySource;
+    private final PropertySource properties;
+    private final AliasManager aliasManager;
+    private JKey genesisKey;
+    private final List<HederaAccount> blockedAccountsCreated = new ArrayList<>();
+
+    @Inject
+    public BlocklistAccountCreator(
+            final Supplier<HederaAccount> accountSupplier,
+            final EntityIdSource ids,
+            final BackingStore<AccountID, HederaAccount> accounts,
+            final Supplier<JEd25519Key> genesisKeySource,
+            final @CompositeProps PropertySource properties,
+            final AliasManager aliasManager) {
+        this.accountSupplier = accountSupplier;
+        this.ids = ids;
+        this.accounts = accounts;
+        this.genesisKeySource = genesisKeySource;
+        this.properties = properties;
+        this.aliasManager = aliasManager;
+    }
+
+    public void ensureBlockedAccounts() {
+        final var blockedEVMAddresses = getPrivateKeyBlocklist().stream()
+                .map(Hex::decode)
+                .map(this::ecdsaPrivateToPublicKey)
+                .map(EthSigsUtils::recoverAddressFromPubKey)
+                .map(ByteString::copyFrom)
+                .filter(evmAddress -> aliasManager.lookupIdBy(evmAddress).equals(MISSING_NUM))
+                .collect(Collectors.toSet());
+
+        for (var evmAddress : blockedEVMAddresses) {
+            final var genesisAccountId = AccountID.newBuilder()
+                    .setRealmNum(0)
+                    .setShardNum(0)
+                    .setAccountNum(2)
+                    .build();
+
+            final var newId = ids.newAccountId(genesisAccountId);
+            if (accounts.contains(newId)) {
+                continue;
```

Comment:
So if the genesis account exists you just go on to the next loop?  Not exit the loop entirely?

Reply:
`ids.newAccountId(genesisAccountId)` is supposed to return the next new sequential account id that is not used, so the check for `accounts.contains(newId)` might be redundant, but just in case we get an existing account id we iterate until we get one that does not exists. Probably I can remove the check for `accounts.contains(newId)` @Neeharika-Sompalli @tinker-michaelj wdyt?

Reply:
but the `continue` is for the loop fetching blocked addresses! it's one of _them_ you're skipping entirely .... (I think?)

Reply:
Ah yes, sorry, I was focusing on the wrong thing, that `continue` is not quite right skipping a blocked address.
I can remove it if the `accounts.contains(newId)` is redundant otherwise I can do something like loop on `ids.newAccountId(genesisAccountId)` until I get an id that is `!accounts.contains(newId)`.

Reply:
Removed the check as it seems redundant.